src/Widgets/SaveDialog.vala

```diff
@@ -158,6 +173,12 @@ public class Screenshot.SaveDialog : Granite.Dialog {
         var save_btn = (Gtk.Button) add_button (_("Save"), Gtk.ResponseType.APPLY);
         save_btn.get_style_context ().add_class (Gtk.STYLE_CLASS_SUGGESTED_ACTION);
 
+        name_entry.changed.connect (() => {
+            name_entry.is_valid = !name_entry.text.contains ("/");
```

Comment:
I would also invalidate the field if the name_entry is empty. When entry, a dotfile `.png` is created, which is pretty unintuitive (and is not visible by default in Files).